### ISSUE TITLE
java.lang.NullPointerException when using -Dtags=""

### ISSUE THREAD

**Issue by yavgel** [serenity-bdd/serenity-core#1940]
`<serenity.version>2.1.1</serenity.version>`

1) having the next Junit test (using the proper `features` structure as described here -> https://serenity-bdd.github.io/theserenitybook/latest/_images/junit-test-dir-structure.png)

```
package features.login;

import base.UiBasePreconditions;
import net.serenitybdd.junit.runners.SerenityRunner;
import net.thucydides.core.annotations.WithTag;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(SerenityRunner.class)
public class LoginPositiveTest extends UiBasePreconditions {

    @Test
    @WithTag("qa_test")
    public void login_positive_test() {
        headerSteps.current_username_widget_should_be_displayed_with_text("William Arthur");
    }
}

```
2) then trying to execute it via next command `cd qa-uk-portal && mvn clean verify -Dtags="qa_test"` 
(I have multiple modules in my project so I have to `cd` first)

**Current results:**

```
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< qa_automation:qa-uk-portal >---------------------
[INFO] Building qa-uk-portal 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ qa-uk-portal ---
[INFO] Deleting /Users/sasha/Downloads/qa_automation/qa-uk-portal/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ qa-uk-portal ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/sasha/Downloads/qa_automation/qa-uk-portal/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ qa-uk-portal ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ qa-uk-portal ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ qa-uk-portal ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 62 source files to /Users/sasha/Downloads/qa_automation/qa-uk-portal/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ qa-uk-portal ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running features.customers.CustomersTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 

-------------------------------------------------------------------------------------
     _______. _______ .______       _______ .__   __.  __  .___________.____    ____ 
    /       ||   ____||   _  \     |   ____||  \ |  | |  | |           |\   \  /   / 
   |   (----`|  |__   |  |_)  |    |  |__   |   \|  | |  | `---|  |----` \   \/   /  
    \   \    |   __|  |      /     |   __|  |  . `  | |  |     |  |       \_    _/   
.----)   |   |  |____ |  |\  \----.|  |____ |  |\   | |  |     |  |         |  |     
|_______/    |_______|| _| `._____||_______||__| \__| |__|     |__|         |__|    
                                                                                     
 News and tutorials at http://www.serenity-bdd.info                                  
 Documentation at https://wakaleo.gitbooks.io/the-serenity-book/content/             
 Join the Serenity Community on Rocket Chat at https://serenity-bdd.rocket.chat      
 Serenity BDD Support and Training at http://serenity-bdd.info/#/trainingandsupport  
 Learn Serenity BDD online at http://serenity-dojo.com                               
-------------------------------------------------------------------------------------

[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Customers tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 1
[pool-1-thread-1] INFO net.thucydides.core.requirements.PackageRequirementsTagProvider - Loading requirements from package requirements at: features
[pool-2-thread-1] INFO net.thucydides.core.requirements.PackageRequirementsTagProvider - Loading requirements from package requirements at: features
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.56 s <<< FAILURE! - in features.customers.CustomersTests
[ERROR] features.customers.CustomersTests  Time elapsed: 0.704 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.report_scheduler.ReportSchedulerTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Report scheduler tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 2
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.333 s <<< FAILURE! - in features.report_scheduler.ReportSchedulerTests
[ERROR] features.report_scheduler.ReportSchedulerTests  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.user_details.UserDetailsTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: User details tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 3
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.203 s <<< FAILURE! - in features.user_details.UserDetailsTests
[ERROR] features.user_details.UserDetailsTests  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.api_3ds_authenticator.CreateAuthenticatorTests
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 s - in features.api_3ds_authenticator.CreateAuthenticatorTests
[INFO] Running features.cards.CardsTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Cards tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 4
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.16 s <<< FAILURE! - in features.cards.CardsTests
[ERROR] features.cards.CardsTests  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.processors.ProcessorsTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Processors tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 5
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.201 s <<< FAILURE! - in features.processors.ProcessorsTests
[ERROR] features.processors.ProcessorsTests  Time elapsed: 0.002 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.transactions.TransactionsTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Transactions tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 6
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.231 s <<< FAILURE! - in features.transactions.TransactionsTests
[ERROR] features.transactions.TransactionsTests  Time elapsed: 0.006 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.organisations.OrganisationsTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Organisations tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 7
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.156 s <<< FAILURE! - in features.organisations.OrganisationsTests
[ERROR] features.organisations.OrganisationsTests  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.accounts.AccountsTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Accounts tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 8
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.19 s <<< FAILURE! - in features.accounts.AccountsTests
[ERROR] features.accounts.AccountsTests  Time elapsed: 0.002 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.transactions_blocked.TransactionsBlockedTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Transactions blocked tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 9
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.432 s <<< FAILURE! - in features.transactions_blocked.TransactionsBlockedTests
[ERROR] features.transactions_blocked.TransactionsBlockedTests  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.user_details_edit.UserDetailsEditTests
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in features.user_details_edit.UserDetailsEditTests
[INFO] Running features.users.UsersTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Users tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 10
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.177 s <<< FAILURE! - in features.users.UsersTests
[ERROR] features.users.UsersTests  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.transaction_refunds.TransactionRefundsTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Transaction refunds tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 11
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.162 s <<< FAILURE! - in features.transaction_refunds.TransactionRefundsTests
[ERROR] features.transaction_refunds.TransactionRefundsTests  Time elapsed: 0.002 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.transactions_tagged.TransactionsTaggedTests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Transactions tagged tests
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: Initialisation
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 12
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.155 s <<< FAILURE! - in features.transactions_tagged.TransactionsTaggedTests
[ERROR] features.transactions_tagged.TransactionsTaggedTests  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NullPointerException

[INFO] Running features.api_card.CardTests
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in features.api_card.CardTests
[INFO] Running features.login.LoginNegativeTests
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 s - in features.login.LoginNegativeTests
[INFO] Running features.login.LoginPositiveTest
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - Test Suite Started: Login positive test
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
 _____ _____ ____ _____   ____ _____  _    ____ _____ _____ ____  
|_   _| ____/ ___|_   _| / ___|_   _|/ \  |  _ \_   _| ____|  _ \ 
  | | |  _| \___ \ | |   \___ \ | | / _ \ | |_) || | |  _| | | | |
  | | | |___ ___) || |    ___) || |/ ___ \|  _ < | | | |___| |_| |
  |_| |_____|____/ |_|   |____/ |_/_/   \_\_| \_\|_| |_____|____/ 
                                                                  

TEST STARTED: login_positive_test
-------------------------------------------------------------------
[main] INFO net.serenitybdd.core.Serenity - TEST NUMBER: 13
Feb 03, 2020 5:18:29 PM org.openqa.selenium.remote.DesiredCapabilities chrome
INFO: Using `new ChromeOptions()` is preferred to `DesiredCapabilities.chrome()`
[main] INFO net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Instantiating driver
[main] INFO net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Driver capabilities: Capabilities {acceptInsecureCerts: false, browserName: chrome, chrome.switches: --headless, goog:chromeOptions: {args: [--headless], extensions: []}, loggingPrefs: org.openqa.selenium.logging..., platform: ANY, version: }
Starting ChromeDriver 79.0.3945.36 (3582db32b33893869b8c1339e8f4d9ed1816f143-refs/branch-heads/3945@{#614}) on port 25472
Only local connections are allowed.
Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
Feb 03, 2020 5:18:30 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - STARTING STEP Open gateway login page
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - FINISHING STEP
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - STARTING STEP Log in with: testuser_automation@dimeboxpayments.com, D1mebox!
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - STARTING STEP Close toast message
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - FINISHING STEP
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - FINISHING STEP
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - STARTING STEP Current username widget should be displayed with text: William Arthur
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - FINISHING STEP
[main] INFO net.thucydides.core.steps.ConsoleLoggingListener - 
        __    _____ _____ ____ _____   ____   _    ____  ____  _____ ____  
  _     \ \  |_   _| ____/ ___|_   _| |  _ \ / \  / ___|/ ___|| ____|  _ \ 
 (_)_____| |   | | |  _| \___ \ | |   | |_) / _ \ \___ \\___ \|  _| | | | |
  _|_____| |   | | | |___ ___) || |   |  __/ ___ \ ___) |___) | |___| |_| |
 (_)     | |   |_| |_____|____/ |_|   |_| /_/   \_\____/|____/|_____|____/ 
        /_/                                                                

TEST PASSED: Login positive test
----------------------------------------------------------------------------
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 13.783 s - in features.login.LoginPositiveTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   AccountsTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   CardsTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   CustomersTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   OrganisationsTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   ProcessorsTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   ReportSchedulerTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   TransactionRefundsTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   TransactionsTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   TransactionsBlockedTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   TransactionsTaggedTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   UserDetailsTests>UiBasePreconditions.logout:86 ? NullPointer
[ERROR]   UsersTests>UiBasePreconditions.logout:86 ? NullPointer
[INFO] 
[ERROR] Tests run: 13, Failures: 0, Errors: 12, Skipped: 0
[INFO] 
[ERROR] There are test failures.

Please refer to /Users/sasha/Downloads/qa_automation/qa-uk-portal/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ qa-uk-portal ---
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] Building jar: /Users/sasha/Downloads/qa_automation/qa-uk-portal/target/qa-uk-portal-1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-failsafe-plugin:2.22.2:integration-test (default) @ qa-uk-portal ---
[INFO] 
[INFO] --- serenity-maven-plugin:2.1.1:aggregate (serenity-reports) @ qa-uk-portal ---
[INFO] Loading requirements from package requirements at: features
[INFO] Test results for 1 tests generated in 1.4 secs in directory: file:/Users/sasha/Downloads/qa_automation/qa-uk-portal/target/site/serenity/
[INFO] -----------------------------------------
[INFO]  SERENITY TESTS : ERROR
[INFO] -----------------------------------------
[INFO] | Tests executed         | 13
[INFO] | Tests passed           | 1
[INFO] | Tests failed           | 0
[INFO] | Tests with errors      | 12
[INFO] | Tests compromised      | 0
[INFO] | Tests pending          | 0
[INFO] | Tests ignored/skipped  | 0
[INFO] ------------------------ | --------------
[INFO] | Total Duration         | 13s 491ms
[INFO] | Fastest test took      | 13s 491ms
[INFO] | Slowest test took      | 13s 491ms
[INFO] -----------------------------------------
[INFO] 
[INFO] SERENITY REPORTS
[INFO]   - Full Report: file:///Users/sasha/Downloads/qa_automation/qa-uk-portal/target/site/serenity/index.html
[INFO] 
[INFO] --- maven-failsafe-plugin:2.22.2:verify (default) @ qa-uk-portal ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  25.498 s
[INFO] Finished at: 2020-02-03T17:18:45+01:00
[INFO] ------------------------------------------------------------------------

```

**Comment by yavgel** [comment]
@wakaleo  - as you can see it's kind of ran through all the features entities with `java.lang.NullPointerException` and only then successfully executed my `LoginPositiveTest` @WithTag("qa_test") - which is strange.
also in the report - | Tests executed | 13 -> which is also wrong.

Please assist - thank you a lot in advance! Love your Serenity !!!

**Comment by wakaleo** [comment]
I can't do much without a stack trace, try running with -e or -X

**Comment by yavgel** [comment]
Here you are @wakaleo 

[log.txt](https://github.com/serenity-bdd/serenity-core/files/4148920/log.txt)

**Comment by wakaleo** [comment]
I'm still not seeing a stack trace - can you provide a project that reproduces the issue?

**Comment by yavgel** [comment]
> I'm still not seeing a stack trace - can you provide a project that reproduces the issue?

https://send.firefox.com/download/5d2e4837c7ef8a96/#FQEErE9eqKa5sw6o4LhPpg

**Comment by yavgel** [comment]
@wakaleo  - Also please note that I also experienced this issue from the most latest Serenity core releases even for 1 module project.

**Comment by wakaleo** [comment]
It's probably related to the new tag processing logic. Tags now support Cucumber tag expressions, but this adds a few constraints. The summary is:
  - Don't use @Step on static methods or with @AfterClass
  - Don't use tags with spaces

**Comment by wakaleo** [comment]
@yavgel could you confirm if these changes fixed your issues?

**Comment by yavgel** [comment]
@wakaleo  - just retested - you were 100% right - `Don't use @step on static methods or with @afterclass` helped me to avoid the original issues. So please feel free to close this issue. Thank you a lot!

**Comment by bensashi** [comment]
> It's probably related to the new tag processing logic. Tags now support Cucumber tag expressions, but this adds a few constraints. The summary is:
> 
> * Don't use @step on static methods or with @afterclass
> * Don't use tags with spaces

I realize this thread is pretty old, but I just ran into the "no tags with spaces" issue in the process of trying to update our Serenity version. Do you happen to know the exact version when the support for tags with spaces was dropped? We're on a very outdated Serenity release, and are only updating at the moment to address a problem with executing tests using function-level (as opposed to class-level) tags. That issue was supposedly fixed as of 2.0.42, so I'm thinking as a short-term solution we might try to find a version equal to or greater than 2.0.42 that still supports tags with spaces, so we don't have to also go update a bunch of tags.
